### PR TITLE
Add documentation page about the fetch service

### PR DESCRIPTION
**Pull request by ineesalmeida**

### CODE REVIEW

File: reference/services/fetch-service.rst [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+   end session ``session ID``, which deletes all data for that session that
+   was stored in the fetch service database.
+
+Using the fetch service
+~~~~~~~~~~~~~~~~~~~~~~~
+Currently, only snaps can use the fetch service.
+
+To do so, a Launchpad admin is required to set the
+``use_fetch_service`` flag within a snap to ``true``, either in the API or in
+the UI by accessing the Admin area of a snap.
```
Comment (jugmac00) [inline reference/services/fetch-service.rst:60]: Admin -> admin

File: reference/services/fetch-service.rst [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+The following repositories also contain code relevant to how Launchpad uses
+the fetch service:
+
+* `buildd <https://git.launchpad.net/~launchpad/launchpad-buildd>`_
+* `launchpad-buildd-manager <https://git.launchpad.net/~launchpad/launchpad/tree/charm/launchpad-buildd-manager>`_
+
+Bug trackers
+------------
+* `Fetch service snap GitHub issues <https://github.com/canonical/fetch-service/issues>`_
+* `Fetch service charm GitHub issues <https://github.com/canonical/fetch-operator/issues>`_
```
Comment (jugmac00) [inline reference/services/fetch-service.rst:94]: Sometimes you capitalize snap and charm, sometimes not. I'd assume the capitalized version is the right one.

File: reference/services/fetch-service.rst [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+Related specifications
+----------------------
+* `LP136 - Integrating the fetch service for snap builds <https://docs.google.com/document/d/1Z2kVh8eGzDV1-zEyTRYbCNQ0fsXJWt9-vutAjZ9Cxck>`_
+* `ST108 - Fetch service control API <https://docs.google.com/document/d/1Ta0THOsHLwbOA6H7ewHa-6s2GtZRWxvvtiMKFk5jiq8>`_
+
+Log files
+---------
+Production
+~~~~~~~~~~
+Not set up.
```
Comment (jugmac00) [inline reference/services/fetch-service.rst:124]: Would IS be able to view the logs? If so, how?

**Review by jugmac00 — CHANGES_REQUESTED** [review]
Thanks a lot for the PR!

The fetch service repos are private, so the link checker fails. You need to add the links to https://github.com/canonical/launchpad-manual/blob/5969eac325484194150235caac48c6e186b55833/custom_conf.py#L112

To lower turnaround time, you can run the checks locally via make, e.g. `make linkcheck`.

File: reference/services/fetch-service.rst (hunk 3 above)
Comment (ineesalmeida) [reply reference/services/fetch-service.rst:124]: That's part of another ticket. I added details here, and added a comment in the ticket to update this section of the documentation

File: reference/services/fetch-service.rst (hunk 2 above)
Comment (ineesalmeida) [reply reference/services/fetch-service.rst:94]: I made it coherent now, I changed it to non-capitalized - I felt it made more sense

**Comment by ineesalmeida** [comment]
Thank you for the review, I've addressed all the comments and the checks are now passing.
I had some trouble getting the links to work though, I think something might be missing. To link between the build-farm and the fetch-service files, I used what is used in other documents:
```
See :doc:`Build Farm <../build-farm>` for more details
```
But when I run it locally, the link doesn't work.

When I do
```
See `Build Farm <../build_farm>`_ for more details
```
instead, the links works but the linkchecker fails. I need to further look into what's missing regarding the links.

**Comment by simonepelosi** [comment]
> Thank you for the review, I've addressed all the comments and the checks are now passing. I had some trouble getting the links to work though, I think something might be missing. To link between the build-farm and the fetch-service files, I used what is used in other documents:
> 
> ```
> See :doc:`Build Farm <../build-farm>` for more details
> ```
> 
> But when I run it locally, the link doesn't work.
> 
> When I do
> 
> ```
> See `Build Farm <../build_farm>`_ for more details
> ```
> 
> instead, the links works but the linkchecker fails. I need to further look into what's missing regarding the links.

I pushed a small fix for that :) the path was wrong, the correct one is <build-farm> instead of <../build-farm>

**Comment by ineesalmeida** [comment]
Thank you Simone! I fixed the remaining links, so it's fully ready now

**Review by jugmac00 — APPROVED** [review]
Thanks a lot! This looks great.

**Pull request merged**